jsonpickle/handlers.py

```diff
@@ -196,11 +197,30 @@ def restore(self, data):
         return cls.__new__(cls, *params)
 
 
-DatetimeHandler.handles(datetime.datetime)
 DatetimeHandler.handles(datetime.date)
 DatetimeHandler.handles(datetime.time)
 
 
+class DatetimeObjectHandler(BaseHandler):
+    """Serialization for datetime.datetime objects"""
+
+    def flatten(self, obj, data):
+        if not self.context.unpicklable:
+            return obj.isoformat()
+        data['iso'] = obj.isoformat()
+        data['tzinfo'] = self.context.flatten(obj.tzinfo, reset=False)
+        return data
+
+    def restore(self, data):
```

Comment:
I forget exactly how restoration works, but I'm pretty sure this change is not backwards or forward compatible. Illustrated by this command, which serializes using the released jsonpickle then attempts to deserialize it with the new version:

```
draft $ pip-run -q jsonpickle -- -c "import jsonpickle, datetime; print(jsonpickle.dumps(datetime.datetime.now()))" | pip-run -q --use-pep517 git+https://github.com/davvid/jsonpickle@isoformat python-dateutil -- -c "import jsonpickle, sys; print(jsonpickle.loads(sys.stdin.read()))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-run-p1td74z2/jsonpickle/unpickler.py", line 41, in decode
    return context.restore(data, reset=reset, classes=classes)
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-run-p1td74z2/jsonpickle/unpickler.py", line 156, in restore
    value = self._restore(obj)
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-run-p1td74z2/jsonpickle/unpickler.py", line 213, in _restore
    return restore(obj)
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-run-p1td74z2/jsonpickle/unpickler.py", line 321, in _restore_object
    instance = handler(self).restore(obj)
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-run-p1td74z2/jsonpickle/handlers.py", line 215, in restore
    obj = dateutil.parser.isoparse(data['iso'])
KeyError: 'iso'
```